src/script.js

```diff
@@ -1,36 +1,68 @@
-const mainObj = {};
-let globalId = 0;
-const form = document.getElementById("input-form");
-
-
-const displayTasks = () => {
-    const tasksListBefore = document.getElementById("tasks-list");
-    tasksListBefore.outerHTML = "<ul id='tasks-list'></ul>";
-    const tasksListAfter = document.getElementById("tasks-list");
-
-    if (Object.keys(mainObj).length > 0)
-        for (const [key, value] of Object.entries(mainObj)) {
-            tasksListAfter.innerHTML += `<li id=${key} style='margin-bottom: 10px;'> ${value} 
-                <button class='list-btn' type='button' onclick="deleteTask(this.parentNode.id)">
-                <img src='delete.svg' alt='Delete Button' width="20px" height="20px"/></button>
-            </li>`;
-        }
+const recipes = [];
+let recipeId = 0;
+let productId = 0;
+
+const form = document.getElementById("recipe-form");
+const body = document.getElementsByTagName("body")[0];
+const taskList = document.getElementById("tasks-list");
+
+
+class Product {
+    constructor(id, name) {
+        this.id = id;
+        this.name = name;
+    }
+}
+
+
+class Recipe {
+    constructor(id, title, products) {
+        this.id = id;
+        this.title = title;
+        this.products = products;
+    }
 }
 
-const createTask = (event) => {
+const createRecipe = (event) => {
     event.preventDefault();
+    recipeId++;
+
+    recipes[recipeId] = new Recipe(recipeId, event.target[0].value, []);
+    event.target[0].value = "";
+}
+
+const removeRecipe = (recipeId) => {
+    delete recipes[recipeId];
+}
+
+const createProduct = (recId) => {
+    const value = document.getElementById(`product-input-${recId}`).value;
+
+    if (value) {
+        productId++;
+        const product = new Product(productId, value);
+
+        recipes[recId].products.push(product);
+    }
+}
+
+const removeProduct = (recId) => {
+    const value = document.getElementById(`product-input-${recId}`).value;
 
-    globalId++;
+    value && (recipes[recId].products = recipes[recId].products.filter(product => product.name !== value));
 
-    mainObj[globalId] = event.target[0].value;
-    return displayTasks();
 }
 
-const deleteTask = (identifier) => {
-    const {id} = document.getElementById(identifier);
-    delete mainObj[id];
-    return displayTasks();
+const refresh = () => {
+    taskList.innerHTML = recipes.map(recipe => `<li id="recipe-${recipe.id}">Tytuł: ${recipe.title} 
+         Produkty: ${recipe.products.length > 0 ? recipe.products.map(product => ` ${product.name}`) : ""}
+         <button type="button" onclick="removeRecipe(${recipe.id}); refresh()">X</button>
+         <input id = "product-input-${recipe.id}" placeholder="Nazwa produktu:"/>
+         <button type="button" onclick="createProduct(${recipe.id}); refresh()">Dodaj produkt</button>
```

Comment:
Total fuck it up. You assign listener here and also on the end of file 
`body.addEventListener("submit", refresh); `
decide on something